### ISSUE TITLE
Incorrect column in no-missing-end-of-source-newline

### ISSUE THREAD

**Issue by Arcanemagus** [stylelint/stylelint#2374]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug where `no-missing-end-of-source-newline` returns the wrong column.

> Which rule, if any, is this issue related to?

`no-missing-end-of-source-newline`

> What CSS is needed to reproduce this issue?

```scss
&.active {
    top:
    .tab {}
}
```

Note that the orphaned `top:` is required to trigger this bug, if it is removed the position returned is correct.

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "extends": [
    "stylelint-config-standard"
  ]
}
```

> Which version of stylelint are you using?

`7.9.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

* CLI with `stylelint --syntax scss main.scss`
* Node.js API through the `linter-stylelint` Atom package

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, it's related to SCSS maps.

> What did you expect to happen?

The correct column (2) to be reported.

> What actually happened (e.g. what warnings or errors you are getting)?

Column 3, which doesn't exist, is reported as the location of the error.

```shell
.\node_modules\.bin\stylelint --syntax scss .\main.scss

main.scss
 2:5  ×  Expected indentation of 2 spaces                                 indentation
 2:9  ×  Expected single space after ":" with a single-line declaration   declaration-colon-space-after
 3:8  ×  Expected a trailing semicolon                                    declaration-block-trailing-semicolon
 3:9  ×  Unexpected whitespace at end of line                             no-eol-whitespace
 4:3  ×  Unexpected missing end-of-source newline                         no-missing-end-of-source-newline
```

Files used to test this can be found here: [scss_no-missing-end-of-source-newline.zip](https://github.com/stylelint/stylelint/files/791942/scss_no-missing-end-of-source-newline.zip)

**Comment by evilebottnawi** [comment]
@Arcanemagus strange on last `postcss-scss` i got error:
```shell
TypeError: this.spacesFromStart is not a function
    at ScssParser.rule (/path/to/node_modules/postcss-scss/lib/scss-parser.js:122:39)
    at ScssParser.other (/path/to/node_modules/postcss/lib/parser.js:141:26)
    at ScssParser.loop (/path/to/node_modules/postcss/lib/parser.js:81:26)
    at scssParse /path/to/node_modules/postcss-scss/lib/scss-parse.js:21:12)
    at new LazyResult (/path/to/node_modules/postcss/lib/lazy-result.js:70:24)
    at Processor.process (/path/to/node_modules/postcss/lib/processor.js:117:12)
    at getCode.then.code (/path/to/node_modules/stylelint/lib/getPostcssResult.js:81:29)
```
/cc @ai regression or expected?

**Comment by ai** [comment]
@evilebottnawi i fixed this issue in `postcss-scss`. Be sure to update for latest version.
https://github.com/postcss/postcss-scss/blob/master/CHANGELOG.md#041

**Comment by evilebottnawi** [comment]
@Arcanemagus column should be `1` (`}`), not `2` (on column `2` no characters)

**Comment by evilebottnawi** [comment]
@ai seems bug in `postcss`, we use `https://github.com/postcss/postcss/blob/master/lib/warning.es6#L41` for line and column.
For code:
```scss
&.active {
    top:
    .tab {}
}
```
`opts`:
```shell
{ severity: 'error',
  rule: 'no-missing-end-of-source-newline',
  node: 
   Root {
     raws: { semicolon: false, after: '' },
     type: 'root',
     nodes: [ [Object] ],
     source: { input: [Object], start: [Object] },
     lastEach: 75,
     indexes: {} },
  index: 32 }
```
But pos return:
```shell
{ line: 4, column: 3 }
```
I try `console.log(opts.node.source.input.css[opts.index]);` before `var pos = opts.node.positionBy(opts);` and get `}` character (it is right).

**Comment by ai** [comment]
@evilebottnawi could you please create a issue in `postcss/postcss`?

**Comment by evilebottnawi** [comment]
@ai I will try to further understand why the column is not returned correctly, them create issue, now it turns out smeared :sob:

**Comment by Arcanemagus** [comment]
@evilebottnawi Ha! Trying to replicate that error is how I found this bug, but I was never able to myself.

**Comment by jeddy3** [comment]
Closing [as stylistic rules are frozen](https://github.com/stylelint/stylelint/issues/5674).
